src/gunicorn.conf.py

```diff
@@ -29,127 +41,222 @@
+
+        # Serialize and store files information in the environment variable (so
+        # workers see it)
+        os.environ["UPLOADED_FILE_MAP"] = json.dumps(files)
+        logger.info(
+            f"Set env UPLOADED_FILE_MAP = {os.environ['UPLOADED_FILE_MAP']}")
+
+        # Create the vector store using the file IDs.
+        vector_store = await ai_client.agents.create_vector_store_and_poll(
+            file_ids=[info["id"] for info in files.values()],
+            name="sample_store"
+        )
+        logger.info("agent: file store and vector store success")
+
+        file_search_tool = FileSearchTool(vector_store_ids=[vector_store.id])
+        toolset.add(file_search_tool)
+
+    return toolset
+
+
+async def create_agent(ai_client: AIProjectClient,
+                       creds: AsyncTokenCredential) -> Agent:
+    logger.info("Creating new agent with resources")
 
     agent = await ai_client.agents.create_agent(
         model=os.environ["AZURE_AI_AGENT_DEPLOYMENT_NAME"],
-        name=os.environ["AZURE_AI_AGENT_NAME"], 
+        name=os.environ["AZURE_AI_AGENT_NAME"],
         instructions="You are helpful assistant",
-        toolset=toolset
+        toolset=await get_available_toolset(ai_client, creds)
     )
     return agent
 
 
-async def update_agent(agent: Agent, ai_client: AIProjectClient) -> Agent:
+async def update_agent(agent: Agent, ai_client: AIProjectClient,
+                       creds: AsyncTokenCredential) -> Agent:
     logger.info("Updating agent with resources")
-    files: Dict[str, Dict[str, str]] = {}
-
-    # Upload files for file search
-    for file_name in FILES_NAMES:
-        file_path = os.path.abspath(os.path.join(os.path.dirname(__file__), 'files', file_name))
-        file = await ai_client.agents.upload_file_and_poll(file_path=file_path, purpose=FilePurpose.AGENTS)
-        # Store both file id and the file path using the file name as key.
-        files[file_name] = {"id": file.id, "path": file_path}
-
-    # Serialize and store files information in the environment variable (so workers see it)
-    os.environ["UPLOADED_FILE_MAP"] = json.dumps(files)
-    logger.info(f"Set env UPLOADED_FILE_MAP = {os.environ['UPLOADED_FILE_MAP']}")
-
-    # Create the vector store using the file IDs.
-    vector_store = await ai_client.agents.create_vector_store_and_poll(
-        file_ids=[info["id"] for info in files.values()],
-        name="sample_store"
-    )
-    logger.info("agent: file store and vector store success")
-
-    file_search_tool = FileSearchTool(vector_store_ids=[vector_store.id])
-    toolset = AsyncToolSet()
-    toolset.add(file_search_tool)
 
     agent = await ai_client.agents.update_agent(
         assistant_id=agent.id,
         model=os.environ["AZURE_AI_AGENT_DEPLOYMENT_NAME"],
         name=os.environ["AZURE_AI_AGENT_NAME"],
         instructions="You are helpful assistant",
-        toolset=toolset
+        toolset=await get_available_toolset(ai_client, creds)
```

Comment:
please separate outside of update_agent and create_agent call

await get_available_toolset(ai_client, creds)

Reply:
Fixed.